README.md

```diff
@@ -1,3 +1,55 @@
+
+To distinguish between the addresses on the trusted logical block device (i.e., MlsDisk) and on the untrusted host block device, we term the former as _logical block addresses (LBAs)_ and the latter _host block addresses (HBAs)_.
+
+The security goal of MlsDisk is to provide to its users (e.g., file systems) the following six security guarantees:
+
+* **Confidentiality** guarantees that the user data submitted by any write is not leaked and thus prevents tampering attacks.
+* **Integrity** promises that the user data returned from any read are genuinely generated by the user and thus prevents snooping attacks.
+* **Freshness** ensures that the user data returned from any read are up-to-date and thus prevents rollback attacks.
+* **Consistency** ensures that all the security guarantees are held despite any accidental crashes or crashing attacks.
+* **Atomicity** promises that all writes before a sync operation are persisted in an all-or-nothing manner.
+* **Irreversibility** promises the sync operation is irreversible regardless whether TEE is online or not.
+
+Prior disk I/O protection solutions only provide a subset of MlsDisk's security guarantees. For example, Linux's [dm-crypt](https://docs.kernel.org/admin-guide/device-mapper/dm-crypt.html) and [dm-integrity](https://docs.kernel.org/admin-guide/device-mapper/dm-crypt.html) only protect confidentiality and integrity, respectively. Although Linux's [dm-verity](https://docs.kernel.org/admin-guide/device-mapper/verity.html) ensures both integrity and freshness, it is read-only. As another example, [Intel SGX Protected File System Library](https://www.intel.com/content/www/us/en/developer/articles/technical/overview-of-intel-protected-file-system-library-using-software-guard-extensions.html) protects confidentiality, integrity, freshness, and consistency, but falls short of atomicity and irreversibility.
+
+## Implementation
+
+MlsDisk is written in Rust. The core of MlsDisk's design resides in [`core/layers`](core/layers), showcasing its multilayered log-structured approach.
+
+MlsDisk is also engineered to be portable across different OSes and TEEs, with platform-specific implementations found in [`core/os`](core/os) and notably [`linux`](linux/) for Linux integration.
+As of this moment, MlsDisk has been integrated into two OSes.
+
+* Integrated into Linux based on [Rust-for-Linux](https://github.com/Rust-for-Linux), which is for use in VM TEEs like [AMD SEV](https://developer.amd.com/sev/) and [Intel TDX](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html).
+* Integrated into [Occlum](https://github.com/occlum/occlum), a Rust library OS, which is for use in [Intel SGX](https://www.intel.com/content/www/us/en/developer/tools/software-guard-extensions/overview.html) enclaves.
+
+For more information about the design of MlsDisk, see [the paper]().
```

Comment:
Instead of providing an empty link, how about 

> We plan release a research paper that describes the design and implementation of MlsDisk in the near future.